android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+This document tries to provide a high level overview of some key differences (and similarities) of developing for Android versus developing for iOS. It is aimed to be a reference for people who are starting to or are considering switching from one mobile platform to another.
+
+# Language
+
+**TL;DR: Kotlin is more convenient, Swift is more powerful**
+
+## Common features and similarities (Kotlin and Swift)
+* Mainly object oriented (classes, inheritance, interface/protocol)
+* Statically typed
+* Type inference with option to specify explicit type
+* Type alias
+* Extensions
+* Generics
+* Optionals
```

Comment:
I think these actually work a little differently under-the-hood between the two languages, but I don't know the details very well and it may not matter much in practice.

Reply:
it does work differently under the hood as the swift optional is actually just an enum. it doesn't lead to much different in practice